src/OrbitGl/App.cpp

```diff
@@ -279,6 +279,23 @@ void OrbitApp::OnTimer(const TimerInfo& timer_info) {
   frame_track_online_processor_.ProcessTimer(timer_info, func);
 }
 
+void OrbitApp::OnSystemMemoryUsage(
+    const orbit_grpc_protos::SystemMemoryUsage& system_memory_usage) {
+  TimerInfo timer_info;
+  timer_info.set_type(TimerInfo::kSystemMemoryUsage);
+  timer_info.set_start(system_memory_usage.timestamp_ns());
+  timer_info.set_end(system_memory_usage.timestamp_ns());
+
+  timer_info.mutable_registers()->Reserve(5);
+  timer_info.add_registers(static_cast<uint64_t>(system_memory_usage.total_kb()));
```

Comment:
Thanks for pointing out this!

We use `kMissingInfo = -1` in `SystemMemoryUsage` to indicate that case that the require information is missing. This is the reason why we have `int64_t` rather than `uint64_t` in the memory size raw values.

Now I changed to use `orbit_api::Encode` to encode the value into `unit_64`. This is the same way we use in value tracking: we use `orbit_api::Encode` to encode `int64_t`, `float` and other types into `uint64_t.`